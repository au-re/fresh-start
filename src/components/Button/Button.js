import styled from "styled-components";

/* The hello world of components: A custom Button */

export default styled.button`
  background: ${(props) => (props.primary ? "palevioletred" : "white")};
  color: ${(props) => (props.primary ? "white" : "palevioletred")};
  font-size: 1em;
  padding: 0.25em 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  cursor: pointer;
  transition: 0.3s all;

  :hover {
    background: ${(props) => (props.primary ? "white" : "")};
    color: palevioletred;
  }
`;
