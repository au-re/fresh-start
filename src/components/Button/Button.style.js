import styled from "styled-components";

/* styles for the Button Component */
export default (componentName) => styled(componentName)`
  width: 100%;
`;
