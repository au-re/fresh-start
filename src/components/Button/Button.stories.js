import { withInfo } from "@storybook/addon-info";
import { storiesOf } from "@storybook/react";
import React, { Fragment } from "react";

import Button from "./Button";

storiesOf("Button", module)

  .add("basic usage", withInfo(`Button without parameters`)(
    () =>
      <Fragment>
        <Button>simple button</Button>
        <Button primary>primary button</Button>
      </Fragment>));
