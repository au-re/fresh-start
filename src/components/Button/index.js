import Button from "antd/lib/button";
import ButtonStyle from "./Button.style";
import React from "react";

/**
 * A custom Button
 *
 */
const MyButton = ({ children, primary, ...rest }) =>
  (<Button {...rest}>{children}</Button>);

export default ButtonStyle(MyButton);
