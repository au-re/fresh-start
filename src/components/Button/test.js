import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";

import Button from "./index";

Enzyme.configure({ adapter: new Adapter() });

describe("Button", () => {

  it("renders without props", () => {
    expect(shallow(<Button />)).toMatchSnapshot();
  });

  it("renders as primary button", () => {
    expect(shallow(<Button primary />)).toMatchSnapshot();
  });

  it("renders without children", () => {
    expect(shallow(<Button><span>test</span></Button>)).toMatchSnapshot();
  });
});
