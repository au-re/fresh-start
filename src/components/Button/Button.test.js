import Enzyme, { shallow } from "enzyme";

import Adapter from "enzyme-adapter-react-16";
import Button from "./index.js";
import React from "react";

Enzyme.configure({ adapter: new Adapter() });

describe("Button", () => {

  it("renders without props", () => {
    expect(shallow(<Button />)).toMatchSnapshot();
  });

  it("renders without children", () => {
    expect(shallow(<Button><span>test</span></Button>)).toMatchSnapshot();
  });
});
