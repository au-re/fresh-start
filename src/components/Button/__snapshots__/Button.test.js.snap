// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without children 1`] = `
<MyButton
  className="sc-bdVaJa UoSJW"
>
  <span>
    test
  </span>
</MyButton>
`;

exports[`Button renders without props 1`] = `
<MyButton
  className="sc-bdVaJa UoSJW"
/>
`;
